File: CarParts/Dto/InvoiceDto/AddInvoiceDto.cs
using CarParts.Shared.Enums;
using Microsoft.AspNetCore.Mvc;

namespace CarParts.Dto.InvoiceDto
{
    public class AddInvoiceDto
    {
        public DateTime Date { get; set; }
        public int Cost { get; set; } = 0;
        public int ClientId { get; set; }
        public bool IsImport { get; set; }
        public string? Notes { get; set; }
        public int Services { get; set; }
        public InvoiceType InvoiceType { get; set; }
        public List<AddPartsToInvoice>? Parts { get; set; }
    }
}

File: CarParts/ClientApp/src/api/Invoice/AddInvoiceDto.ts
class AddInvoiceDto {
  date: string|Date = "";
  coast: number = 0;
  clientId: string = '';
  isImport: boolean = false;
  received = true;
  notes: string = "";
  services: number = 0;
  parts: Part[] = [];
}

interface Part {
  partId: number;
  storeId: number;
  quantity: number;
  price: number;
}

export { AddInvoiceDto };
